src/admin/content-block/helpers/generators/image-grid.ts

```diff
@@ -0,0 +1,122 @@
+import { MultiRangeProps } from '@viaa/avo2-components/dist/components/MultiRange/MultiRange';
+import { GridItem } from '@viaa/avo2-components/dist/content-blocks/BlockGrid/BlockGrid';
+
+import { FileUploadProps } from '../../../../shared/components/FileUpload/FileUpload';
+import i18n from '../../../../shared/translations/i18n';
+import { IMAGE_GRID_FILL_OPTIONS, IMAGE_GRID_TEXT_ALIGN_OPTIONS } from '../../content-block.const';
+import {
+	ContentBlockBackgroundColor,
+	ContentBlockConfig,
+	ContentBlockEditor,
+	ContentBlockType,
+	ImageGridBlockComponentStateBlockFields,
+	ImageGridBlockComponentStateFields,
+} from '../../content-block.types';
+
+import {
+	CONTENT_BLOCK_FIELD_DEFAULTS,
+	FILE_FIELD,
+	FORM_STATE_DEFAULTS,
+	TEXT_FIELD,
+} from './defaults';
+
+export const INITIAL_IMAGE_GRID_BLOCK_COMPONENT_STATES = (): ImageGridBlockComponentStateFields[] => [
+	{
+		source: undefined,
```

Comment:
Ik weet niet in hoeverre het nuttig is om iets te verplichten maar undefined dan wel toe te laten. Dit lijkt me persoonlijk geen best practice omdat het gewoon onlogisch klinkt. Wat is je mening hierover?

Reply:
moet inderdaad ingevuld zijn, ik zal het verplicht maken en de default casten naar any